src/components/Layout/Header.tsx

```diff
@@ -24,11 +27,109 @@ const onRightClickLogo = (event: React.MouseEvent): void => {
     }
 }
 
+const navLinks = [
+    {
+        section: 'Product',
+        items: [
+            {
+                title: 'Code Search',
+                href: '/code-search',
```

Comment:
@bretthayes I agree with your suggestion, we can reduce the repeated code by mapping over a data structure. I went with an array here. At this point, I feel as though we could incorporate two components: one for the mobile navbar and the other for the larger navbar. I suspect this would allow us to remove `initialMobileMenuState` in lieu of a simpler useState pattern. Although, in light of our discussion on the FeatureWalkthrough, I would appreciate your feedback first so I don't jump the gun 😅 

Due to the rendering nuances of the `right-nav` links, I feel they would be easier to maintain outside of the pattern used for the `left-nav`. Let me know what you think!

Reply:
This is awesome Tim! Such a great improvement compared to the previous! I agree, and in this case, I think separating the mobile nav and desktop nav makes sense since it would simplify the logic even further. 😄